### PR TITLE
[WIP] Net Standard 2.0 and Should Match Approved

### PR DESCRIPTION
**Pull request by dennisroche**
A work in progress to see what is needed to get `ShouldMatchApproved` working under `netstandard2.0`

- [x] Added `netstandard2.0` to csproj
- [x] Enabled `ShouldMatchApproved` define
- [ ] Fix compile errors :boom:
- [ ] Cross-platform diff tool support

Interesting side note, I have `ShouldMatchApproved` working in `netstandard1.6` by removing the `StackFrame` walking.

### CODE REVIEW

**Comment by JosephWoodward** [comment]
Nice! Was also interested to see if we could get it working on Mac or other non-Windows devices by checking common locations for diffing tools.

**Comment by dennisroche** [comment]
I've added cross-platform diff tool to the checklist

File: src/Shouldly/ShouldlyConfiguration.cs [hunk 1]
```diff
@@ -44,20 +45,20 @@ static ShouldlyConfiguration()
...
         {
-            CallContext.LogicalSetData("ShouldlyDisableSourceInErrors", true);
+            ShouldlyDisableSourceInErrors.Value = true;
             return new EnableSourceInErrorsDisposable();
         }
 
         public static bool IsSourceDisabledInErrors()
         {
-            return (bool?) CallContext.LogicalGetData("ShouldlyDisableSourceInErrors") == true;
+            return ShouldlyDisableSourceInErrors.Value == true;
```
Comment (JosephWoodward) [inline src/Shouldly/ShouldlyConfiguration.cs]: What's the plan with `ShouldlyDisableSourceInErrors`? I can't see it anywhere in your branch? Is this part WIP? I noticed [this article](http://www.cazzulino.com/callcontext-netstandard-netcore.html) that can achieve a polyfill for `CallContext` on non-Windows platforms.

**Comment by JosephWoodward** [comment]
Are you able to give me permission to contribute to this PR? I'm thinking we get it merged (.NET Standard 2.0 support would be good) and add the Cross-platform diff tool support as a separate issue.

**Comment by dennisroche** [comment]
@JosephWoodward you should already have permission to edit the branch. let me know if you don't. apologies for not getting back to finish off the pull request, life has been busy.

**Comment by JosephWoodward** [comment]
@dennisroche I tried to push the changes to your branch but it said I needed permission, apparently there's a checkbox somewhere to grant me privileges? Though I do remember reading recently that it should be checked by default now so perhaps I'm doing something wrong.

File: src/Shouldly/Configuration/VisualStudioDiffTool.cs [hunk 2]
```diff
@@ -50,6 +50,7 @@ private static string GetPath()
                 return null;
             }
 
+#if !NETSTANDARD2_0
             if (process != null)
```
Comment (dennisroche) [inline src/Shouldly/Configuration/VisualStudioDiffTool.cs]: Not sure what to do with `ProcessModule` missing `FileName` property in `netstandard20`

**Comment by JosephWoodward** [comment]
I hope you don't mind but I've upgraded Shouldly to be .NET Standard 2.0 compatible. It would be awesome to get ShouldMatchApproved working and the diff tooling working cross platform though. I'll close this PR for the interim.

**Pull request closed without merging**